Source/Core/Core/HW/EXI/BBA/BuiltIn.cpp

```diff
@@ -0,0 +1,686 @@
+    ref->bba_mac = Common::BitCastPtr<Common::MACAddress>(&m_eth_ref->mBbaMem[BBA_NAFR_PAR0]);
+    ref->my_mac = m_fake_mac;
+    ref->tcp_socket.setBlocking(false);
+
+    // reply with a sin_ack
+    Common::TCPPacket result(ref->bba_mac, ref->my_mac, ref->from, ref->to, ref->seq_num,
+                             ref->ack_num, TCP_FLAG_SIN | TCP_FLAG_ACK);
+
+    result.tcp_options = std::vector<u8>{0x02, 0x04, 0x05, 0xb4, 0x01, 0x01, 0x01, 0x01};
+
+    ref->seq_num++;
+    target = sf::IpAddress(ntohl(Common::BitCast<u32>(ipdata.destination_addr)));
+    ref->tcp_socket.connect(target, ntohs(tcpdata.destination_port));
+    ref->ready = false;
+    ref->ip = Common::BitCast<u32>(ipdata.destination_addr);
+
+    ref->tcp_buffers[0].data = result.Build();
+    ref->tcp_buffers[0].seq_id = ref->seq_num - 1;
+    ref->tcp_buffers[0].tick = GetTickCountStd() - 900;  // delay
+    ref->tcp_buffers[0].used = true;
+  }
+  else
+  {
+    // data packet
+    if (ref == nullptr)
+      return;  // not found
+
+    const int size = ntohs(ipdata.total_len) - ipdata.DefinedSize() - tcpdata.GetHeaderSize();
+    const u32 this_seq = ntohl(tcpdata.sequence_number);
+
+    if (size > 0)
+    {
+      // only if data
+      if (static_cast<int>(this_seq - ref->ack_num) >= 0 && data.size() >= size)
+      {
+        ref->tcp_socket.send(data.data(), size);
+        ref->ack_num += size;
+      }
+
+      // send ack
+      WriteToQueue(BuildAckFrame(ref));
+    }
+    // update windows size
+    ref->window_size = ntohs(tcpdata.window_size);
+
+    // clear any ack data
+    if (ntohs(tcpdata.properties) & TCP_FLAG_ACK)
+    {
+      const u32 ack_num = ntohl(tcpdata.acknowledgement_number);
+      for (auto& tcp_buf : ref->tcp_buffers)
+      {
+        if (!tcp_buf.used || tcp_buf.seq_id >= ack_num)
+          continue;
+
+        Common::PacketView view(tcp_buf.data.data(), tcp_buf.data.size());
+        auto tcp_packet = view.GetTCPPacket();  // This is always a tcp packet
+        if (!tcp_packet.has_value())            // should never happen but just in case
+          continue;
+
+        const u32 seq_end = static_cast<u32>(tcp_buf.seq_id + tcp_packet->data.size());
+        if (seq_end <= ack_num)
+        {
+          tcp_buf.used = false;  // confirmed data received
+          if (!ref->ready && !ref->tcp_buffers[0].used)
+            ref->ready = true;
+          continue;
+        }
+        // partial data, adjust the packet for next ack
+        const u16 ack_size = ack_num - tcp_buf.seq_id;
+        tcp_packet->data.erase(tcp_packet->data.begin(), tcp_packet->data.begin() + ack_size);
+
+        tcp_buf.seq_id += ack_size;
+        tcp_packet->tcp_header.sequence_number = htonl(tcp_buf.seq_id);
+        tcp_buf.data = tcp_packet->Build();
+      }
+    }
+  }
+}
+
+/// <summary>
+/// This is a litle hack, Mario Kart open some UDP port
+/// and listen to it. We open it on our side manualy.
+/// </summary>
```

Comment:
AFAIK, we don't use this style of comment but [regular //](https://github.com/dolphin-emu/dolphin/blob/master/Contributing.md#general) instead.
```C++
// This is a little hack, some games open a UDP port
// and listen to it. We open it on our side manually.
```